src/main/java/database/ListingDatabaseController.java

```diff
@@ -0,0 +1,74 @@
+package database;
+
+import features.Listing;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ListingDatabaseController implements ListingDatabaseGateway, DatabaseInteractor{
+
+    private List<Listing> listings;
+
+    public ListingDatabaseController(String keyword) {
+        this.listings = getListingWithSearch(keyword);
+    }
+
+    public ListingDatabaseController() {
+        this.listings = getListingDefault();
+    }
+
+    @Override
+    public List<Listing> getListingWithSearch(String keyword) {
+        List<Listing> listings = new ArrayList<>();
+        // this will go through database and retrieve first 10 listings with keyword in the listing name
```

Comment:
call (SearchForm instance).getResponseModel() and work with the response model that has a .getListings() method for a array list of listings that fit the query.